src/Commands/DrushCommands.php

```diff
@@ -65,6 +66,18 @@ protected function io(): DrushStyle
         return $this->io;
     }
 
+    /**
+     * Sets a logger, if none is available yet.
+     *
+     * @param LoggerInterface $logger
+     */
+    public function setLoggerIfEmpty(LoggerInterface $logger): void
```

Comment:
Is this used by anything other than the service manager? If not, I think it would be better to just move the condition to `inflect`, and omit the extra trivial function in the base class.

Reply:
We can't do that, the logger variable is protected and can't be accessed from the service manager.

Reply:
what about making logger() public and using it here? maybe only on 13.x branch due to BC.

```
if (!$object->logger()) {
  $object->setLogger($container->get('logger'))
}
```

Reply:
You can tell whether the logger variable is set via https://github.com/drush-ops/drush/blob/95e0acf9ac0cb2a235e217c31d46b048f08a5536/src/Commands/DrushCommands.php#L84

Reply:
That method is protected, so we can't call it from outside, i.e. from the service manager.